### ISSUE TITLE
Conda packages hang

### ISSUE THREAD

**Issue by lparsons** [joshuagryphon/plastid#29]
It seems there is an issue with the regular expression here: https://github.com/joshuagryphon/plastid/blob/master/plastid/util/io/openers.py#L253

When conda packages are built and tested, the paths can be rather long and it seems that this is hanging. I'm not quite sure what the intent of the code is, but I suspect it could be simplified using some `os.path` functions.

**Comment by joshuagryphon** [comment]
Curiouser and curiouser. Thank you for pointing this out. This function is basically used in prettyprints & debugging, so it's really not essential, although convenient.

 I can do some poking around to fix this, but it'll be a few days

**Comment by joshuagryphon** [comment]
Now that the fix in #30 is merged (as of 5be1317b779e39bb9f49d4145fa4a8d4d7c1e393), can you pull latest `develop` and re-test?

**Comment by lparsons** [comment]
I was able to test a build with the develop version, but ran into a new error: `undefined symbol: compress`. I'm not sure if the fix should be in plastid or the bioconda build setup yet.

```
14:32:39 BIOCONDA INFO (OUT) Executing transaction: ...working... done
14:32:40 BIOCONDA INFO (ERR) Traceback (most recent call last):
14:32:40 BIOCONDA INFO (ERR)   File "/opt/conda/conda-bld/plastid_1561127052601/test_tmp/run_test.py", line 2, in <module>
14:32:40 BIOCONDA INFO (ERR)     import plastid
14:32:40 BIOCONDA INFO (ERR)   File "/opt/conda/conda-bld/plastid_1561127052601/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/plastid/__init__.py", line 72, in <module>
14:32:40 BIOCONDA INFO (ERR)     from plastid.readers.bigbed import BigBedReader
14:32:40 BIOCONDA INFO (ERR) ImportError: /opt/conda/conda-bld/plastid_1561127052601/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/plastid/readers/bigbed.cpython-37m-x86_64-linux-gnu.so: undefined symbol: compress
```

**Comment by lparsons** [comment]
I'm not sure if you changed anything regarding how the kentUtils are built, but perhaps that's an issue. I've had [these lines in the `build.sh` script](https://github.com/bioconda/bioconda-recipes/blob/e77e99accde612504b02fbd67fa662fe2d65eb84/recipes/plastid/build.sh#L3-L6) which has worked in previous versions: 
```
# For zlib: https://bioconda.github.io/troubleshooting.html#zlib
export CFLAGS="-I$PREFIX/include"
export LDFLAGS="-L$PREFIX/lib"
export CPATH=${PREFIX}/include
```

**Comment by joshuagryphon** [comment]
I'll take a look. I didn't make any changes to the build system, and everything built fine on my end under Python 2.7, 3.3, 3.6, and 3.7 in my system (Ubuntu 18.04 / in `virtualenvs` managed by `tox`). 

Tests completed under 2.7, 3.3, and 3.6. Under 3.7, they started, but halted mid-way due to #31

**Comment by lparsons** [comment]
I'm wondering if you need to add the line: ` libraries=LIBRARIES + ["z"],` to the `bigbed` Extensions (https://github.com/joshuagryphon/plastid/blob/4e1ded25ad8c285bf198657c74805ddb162a0170/setup.py#L293).

**Comment by joshuagryphon** [comment]
Following up- since we merged #32 (and now since iterators are fixed under Python 3.7), are we able to close this comment out? Are `conda` packages no longer hanging?

**Comment by lparsons** [comment]
With #32 merged, I believe we can close this, yes. Thanks @joshuagryphon.
